### ISSUE TITLE
InfluxDB panic when deleting certain measurement

### ISSUE THREAD

**Issue by scottc-adestra** [influxdata/influxdb#9731]
### Bug report

__System info:__ InfluxDB v1.5.2, CentOS Linux release 7.4.1708 (Core) 

__Steps to reproduce:__

1. Attempt to delete/drop specific measurement

__Expected behavior:__ All data from measurement deleted

__Actual behavior: InfluxDB panics, leaving a number of .tombstone files around which need to be removed before InfluxDB will start again

__Additional info:__ 

This is data from the telegraf log parser plugin. influx_inspect verify checks out and the DB itself seems to be absolutely fine until I attempt to delete this measurement

Log

> influxd: panic: runtime error: slice bounds out of range
influxd: goroutine 75308035 [running]:
influxd: github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Tombstoner).readTombstoneV4.func1(0xc4b9e7abc8, 0xc485dd0580, 0xc4b9e7abf0, 0xc4b9e7abd8, 0xc4b9e7ab90, 0xc4b9e7ab98, 0xc4b9e7add8, 0x42c926, 0xe23db8)
influxd: /go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/tombstone.go:619 +0x5c5
influxd: github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Tombstoner).readTombstoneV4(0xc4201097a0, 0xc42022cdd8, 0xc4b9e7add8, 0x0, 0x0)
influxd: /go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/tombstone.go:638 +0x452
influxd: github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Tombstoner).Walk(0xc4201097a0, 0xc4b9e7add8, 0x0, 0x0)
influxd: /go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/tombstone.go:234 +0x29a
influxd: github.com/influxdata/influxdb/tsdb/engine/tsm1.(*TSMReader).applyTombstones(0xc4223a83c0, 0x0, 0x0)
influxd: /go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/reader.go:259 +0x17a
influxd: github.com/influxdata/influxdb/tsdb/engine/tsm1.(*batchDelete).Commit(0xc42022cdc0, 0x0, 0x0)
influxd: /go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/reader.go:656 +0xb1
influxd: github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).deleteSeriesRange.func1(0x12da060, 0xc4223a83c0, 0xc4223a83c0, 0x0)
influxd: /go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:1377 +0x5cf
influxd: github.com/influxdata/influxdb/tsdb/engine/tsm1.(*FileStore).Apply.func1(0xc4a56ad4a0, 0xc45b810000, 0xc4843b8870, 0x12da060, 0xc4223a83c0)
influxd: /go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/file_store.go:371 +0xc4
influxd: created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*FileStore).Apply
influxd: /go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/file_store.go:365 +0xfc
systemd: influxdb.service: main process exited, code=exited, status=2/INVALIDARGUMENT

**Comment by kmoppel** [comment]
I get the same on 1.5.1 (Ubuntu 16.04) when issuing
```
> delete where time < '2018-04-13'
ERR: %!s(<nil>)
```
Journactl says:
```
apr   30 12:11:16 e5470 influxd[9526]: panic: runtime error: slice bounds out of range
apr   30 12:11:16 e5470 systemd[1]: influxdb.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
```
It kind of a serious problem...running out of disk space but can't clean up data:(

**Comment by benbjohnson** [comment]
@kmoppel I'm not able to reproduce locally. Can you provide either a script to generate ingest data that causes the issue or provide the tombstone files themselves?

**Comment by kmoppel** [comment]
@benbjohnson hi! sorry for lag - yes I still the tombstone file around, but a question first - does it contain any data fields also or only some row ID's that should not be compacted? How should I deliver the file?

**Comment by benbjohnson** [comment]
@kmoppel The tombstone file will contain the series keys and a Unix time range key deletion. How large is the file?

**Comment by kmoppel** [comment]
oh it's small, about 100K..but wouldn't want to make it public still:)

**Comment by benbjohnson** [comment]
@kmoppel Can you send me an e-mail at ben@influxdata.com and I'll share some upload information?

**Comment by kmoppel** [comment]
Tombstone file uploaded. The server config was by the way largely at defaults, except:
```
    sed -i 's/\# query-log-enabled = true/query-log-enabled = false/' /etc/influxdb/influxdb.conf \
    && sed -i 's/\# \[monitor\]/\[monitor\]/' /etc/influxdb/influxdb.conf \
    && sed -i 's/\# store-enabled = true/store-enabled = false/' /etc/influxdb/influxdb.conf \
    && sed -i 's/\# \[http\]/\[http\]/' /etc/influxdb/influxdb.conf \
    && sed -i '0,/\# log-enabled = true/{s/\# log-enabled = true/log-enabled = false/}' /etc/influxdb/influxdb.conf
```

**Comment by benbjohnson** [comment]
Thanks @kmoppel! I'll take a look. 👌

**Comment by benbjohnson** [comment]
@kmoppel Thanks again for the data file. This issue is fixed in `master` but I backported it to `1.5` and will be available in the `1.5.4` release: https://github.com/influxdata/influxdb/pull/9924
